### PR TITLE
RD-3027 Reorganized directory structure in widgets/common

### PR DESCRIPTION
**Pull request by kubama**
## Description
New subdirectories created:
- `actions` - `*Actions` classes that don't fit in any other domnain-specific subdirectory, available through `Stage.Common.Actions`
- `blueprints` - blueprints related files, available through `Stage.Common.Blueprints`
- `components` - common generic React components that don't fit in any other domnain-specific subdirectory, available through `Stage.Common.Components`
- `deployments` - deployments related files that were not already in `deploymentsView` or `deployModal` dirs, available through `State.Common.Deployments`
- `executions` - executions related files, available through `Stage.Common.Executions`
- `inputs` - inputs related files, available through `Stage.Common.Inputs`
- `nodes` - nodes related files, most of them were removed from `Stage.Common` as they are only imported by other files from `common` 
- `plugins` - plugins related files, available through `Stage.Common.Plugins`
- `roles` - roles related files, available through `Stage.Common.Roles`
- `secrets` - secrets related files
- `utils` - generic utilities not fitting into any other subridectory

## Screenshots / Videos
N/A

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/2288/ - PASS (except test that are failing on `master` ATM)

 

## Documentation
N/A

### CODE REVIEW

**Review by qooban — COMMENTED** [review]
I went over the description, and I like the new directory structure very much 👍🏻 Looks like a great improvement!

@kubama I see there are quite a few conflicts to be resolved, so I'll postpone my review until you manage to fix it. I hope it's fine. Let me know if you'd like to get feedback from me earlier.

**Review by jawamartus — COMMENTED** [review]
Partially reviewed. I didn't have a time to take a look of every file.

I left a few comments.

I noticed that system tests and CI pipeline is failing.

Please request my rereview when the tests would be passing and after addressing the comments I left.

File: widgets/common/src/components/parameter/ParameterValue.tsx [hunk 1]
```diff
@@ -3,10 +3,10 @@ import { types } from 'cloudify-ui-common';
 import highlighterStyles from 'react-syntax-highlighter/dist/esm/styles/hljs/idea';
 
 /**
- * ParameterValue is a component which shows parameters (e.g. deployment/blueprint inputs, outputs, runtime properties, ...)
+ * ParameterValue is a component which shows parameter (e.g. deployment/blueprint inputs, outputs, runtime properties, ...)
  * in nice user-friendly formatted manner with copy to clipboard button.
  */
-export default class ParameterValue extends React.Component {
+export default class ParameterValue extends React.Component<{ value: any }> {
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/components/parameter/ParameterValue.tsx]: I added `showCopyButton` to props type. Regarding further changes - they are not related to the PR topic, so I'd rather stop at this point as this PR is already enormous.

File: widgets/common/src/nodes/NodeInstancesSummary.tsx [hunk 2]
```diff
@@ -1,4 +1,6 @@
 // @ts-nocheck File not migrated fully to TS
```
Comment (qooban) [inline widgets/common/src/nodes/NodeInstancesSummary.tsx:1]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

File: widgets/common/src/roles/RolesPicker.tsx [hunk 3]
```diff
@@ -42,8 +42,3 @@ RolesPicker.propTypes = {
     resourceName: PropTypes.string.isRequired,
     toolbox: Stage.PropTypes.Toolbox.isRequired
 };
-
```
Comment (qooban) [inline widgets/common/src/roles/RolesPicker.tsx:45]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

File: widgets/common/src/roles/RolesPresenter.tsx [hunk 4]
```diff
@@ -48,8 +48,3 @@ RolesPresenter.defaultProps = {
     groupRoles: {},
```
Comment (qooban) [inline widgets/common/src/roles/RolesPresenter.tsx:48]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

File: widgets/deployments/src/ActionsMenus.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 // @ts-nocheck File not migrated fully to TS
```
Comment (qooban) [inline widgets/deployments/src/ActionsMenus.tsx:1]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

File: widgets/labels/src/DeleteConfirmModal.tsx [hunk 6]
```diff
@@ -3,7 +3,7 @@ import _ from 'lodash';
```
Comment (qooban) [inline widgets/labels/src/DeleteConfirmModal.tsx:3]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

File: widgets/managers/src/ExecuteWorkflowButton.tsx [hunk 7]
```diff
@@ -2,7 +2,7 @@
```
Comment (qooban) [inline widgets/managers/src/ExecuteWorkflowButton.tsx:2]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

File: widgets/managers/src/ExecuteWorkflowIcon.tsx [hunk 8]
```diff
@@ -1,10 +1,10 @@
 // @ts-nocheck File not migrated fully to TS
```
Comment (qooban) [inline widgets/managers/src/ExecuteWorkflowIcon.tsx:2]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

File: widgets/pluginUploadButton/src/UploadButton.tsx [hunk 9]
```diff
@@ -21,7 +21,7 @@ export default class UploadButton extends React.Component {
         const { toolbox } = this.props;
```
Comment (qooban) [inline widgets/pluginUploadButton/src/UploadButton.tsx:21]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

File: widgets/tenants/src/UserRoles.tsx [hunk 10]
```diff
@@ -16,7 +16,7 @@ export default function UserRoles({ tenant, user }) {
         return roles;
     }
```
Comment (qooban) [inline widgets/tenants/src/UserRoles.tsx:18]: Optional:

As the file is small - maybe try migrating it at once? (only if it is quick)

**Review by qooban — DISMISSED** [review]
Very well done! I appreciate the hard work and as I've mentioned before - the new structure is good and I'm sure the whole UI team will benefit from it. Also, great idea to remove `Stage.Common` uses from `widgets/common/src` at once 👍🏻 

Despite I added 18 comments, there are no blockers from my end, so I'm approving the PR right now. Most of my comments are optional and are related to TS migration. I marked small files (in most cases less than 100 LOCs) which you can consider to migrate at once. I hope you'll be able to do that at once or maybe in follow-up PR (your decision).

File: widgets/common/src/executeWorkflow/ExecuteWorkflowInputs.tsx [hunk 11]
```diff
@@ -1,11 +1,9 @@
...
-import type {
-    OnDateInputChange,
-    OnDropdownChange,
-    OnCheckboxChange,
-    BaseWorkflowInputs,
-    UserWorkflowInputsState
-} from './types';
+import InputsHelpIcon from '../inputs/InputsHelpIcon';
+import getInputFields from '../inputs/utils/getInputFields';
+import { OnChange } from '../inputs/utils/types';
```
Comment (NorbertAmartus) [inline widgets/common/src/executeWorkflow/ExecuteWorkflowInputs.tsx:4]:
```suggestion
import type { OnChange } from '../inputs/utils/types';
```

File: widgets/common/src/executeWorkflow/ExecuteWorkflowModal.tsx [hunk 12]
```diff
@@ -1,21 +1,25 @@
 import type { FunctionComponent } from 'react';
+import FileActions from '../actions/FileActions';
+import DeploymentActions from '../deployments/DeploymentActions';
+import getInputFieldInitialValue from '../inputs/utils/getInputFieldInitialValue';
+import getUpdatedInputs from '../inputs/utils/getUpdatedInputs';
+import { OnChange } from '../inputs/utils/types';
```
Comment (NorbertAmartus) [inline widgets/common/src/executeWorkflow/ExecuteWorkflowModal.tsx:6]:
```suggestion
import type { OnChange } from '../inputs/utils/types';
```

**Review by NorbertAmartus — COMMENTED** [review]
Great one! 🎉 

Added few suggestions, when it comes to importing the types
But besides that, everything looks good to me 💪

File: widgets/common/src/actions/FileActions.ts [hunk 13]
```diff
@@ -1,7 +1,5 @@
 // @ts-nocheck File not migrated fully to TS
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/actions/FileActions.ts]: Done

File: widgets/common/src/components/GroupState.tsx [hunk 14]
```diff
@@ -1,7 +1,5 @@
 // @ts-nocheck File not migrated fully to TS
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/components/GroupState.tsx]: Done

File: widgets/common/src/components/RevertToDefaultIcon.tsx [hunk 15]
```diff
@@ -69,7 +69,4 @@ RevertToDefaultIcon.defaultProps = {
     style: {}
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/components/RevertToDefaultIcon.tsx]: Done

File: widgets/common/src/components/parameter/ParameterValue.tsx [hunk 16]
```diff
@@ -3,10 +3,10 @@ import { types } from 'cloudify-ui-common';
 import highlighterStyles from 'react-syntax-highlighter/dist/esm/styles/hljs/idea';
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/components/parameter/ParameterValue.tsx:3]: It woould be good to migrate it to functional component first, so I'm leaving it for now.

File: widgets/common/src/components/parameter/ParameterValueDescription.tsx [hunk 17]
```diff
@@ -59,8 +59,3 @@ export default function ParameterValueDescription() {
         />
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/components/parameter/ParameterValueDescription.tsx:59]: Done

File: widgets/common/src/deployments/DeploymentUpdatesActions.ts [hunk 18]
```diff
@@ -1,7 +1,5 @@
 // @ts-nocheck File not migrated fully to TS
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/deployments/DeploymentUpdatesActions.ts]: Done

File: widgets/common/src/deploymentsView/map/container.tsx [hunk 19]
```diff
@@ -1,5 +1,8 @@
 import { ComponentProps, FunctionComponent, useMemo } from 'react';
 import { useQuery } from 'react-query';
+import NoDataMessage from '../../components/NoDataMessage';
+import DeploymentActions from '../../deployments/DeploymentActions';
+import MapsActions from '../../map/MapsActions';
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/deploymentsView/map/container.tsx:5]: No, I only changed it in places where I was doing edits

File: widgets/common/src/filters/inputs/AttributeValueInput.tsx [hunk 20]
```diff
@@ -1,7 +1,7 @@
 // @ts-nocheck File not migrated fully to TS
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/filters/inputs/AttributeValueInput.tsx:1]: I found it non trivial so I decided to leave it for now

File: widgets/common/src/inputs/utils/errors.ts [hunk 21]
```diff
@@ -0,0 +1,16 @@
+export function addErrors(inputsWithoutValues: Record<string, any>, errors: Record<string, any>) {
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/inputs/utils/errors.ts:1]: The assumption is correct.

File: widgets/common/src/nodes/NodeInstancesSummary.tsx (hunk 2 above)
Comment (kubama) [reply widgets/common/src/nodes/NodeInstancesSummary.tsx]: Done

File: widgets/common/src/roles/RolesPicker.tsx (hunk 3 above)
Comment (kubama) [reply widgets/common/src/roles/RolesPicker.tsx:45]: Done

File: widgets/common/src/roles/RolesPresenter.tsx (hunk 4 above)
Comment (kubama) [reply widgets/common/src/roles/RolesPresenter.tsx]: Done

File: widgets/deployments/src/ActionsMenus.tsx (hunk 5 above)
Comment (kubama) [reply widgets/deployments/src/ActionsMenus.tsx]: Done

File: widgets/labels/src/DeleteConfirmModal.tsx (hunk 6 above)
Comment (kubama) [reply widgets/labels/src/DeleteConfirmModal.tsx:2]: Done

File: widgets/managers/src/ExecuteWorkflowButton.tsx (hunk 7 above)
Comment (kubama) [reply widgets/managers/src/ExecuteWorkflowButton.tsx:2]: Done

File: widgets/managers/src/ExecuteWorkflowIcon.tsx (hunk 8 above)
Comment (kubama) [reply widgets/managers/src/ExecuteWorkflowIcon.tsx:2]: Done

File: widgets/pluginUploadButton/src/UploadButton.tsx (hunk 9 above)
Comment (kubama) [reply widgets/pluginUploadButton/src/UploadButton.tsx:21]: It woould be good to migrate it to functional component first, so I'm leaving it for now.

**Comment by jawamartus** [comment]
FYI I was merging the changes in DynamicDropdown and InputsUtils today.

They are described here: https://github.com/cloudify-cosmo/cloudify-stage/pull/1887/files
and here: https://github.com/cloudify-cosmo/cloudify-stage/pull/1884

It may be helpful with resolving the conflicts.

File: widgets/tenants/src/UserRoles.tsx (hunk 10 above)
Comment (kubama) [reply widgets/tenants/src/UserRoles.tsx:25]: Done

**Review by NorbertAmartus — DISMISSED** [review]

**Review by qooban — DISMISSED** [review]
Thanks for the additional effort on TS migration 👍🏻

File: widgets/labels/src/DeleteConfirmModal.tsx [hunk 22]
```diff
@@ -1,9 +1,28 @@
-// @ts-nocheck File not migrated fully to TS
 import _ from 'lodash';
 
-export default function DeleteConfirmModal({ deploymentId, labels, labelToDelete, onHide, toolbox }) {
+interface Label {
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/labels/src/DeleteConfirmModal.tsx]: Done

**Review by qooban — APPROVED** [review]

**Review by jawamartus — APPROVED** [review]
I have some concerns that the changes may break some existing modules outiside of `cloudify-stage`. However I understand that it was written in the documentation that Common API may change and because of that it's acceptable to change it.

I checked the places where you have a conflicts with my merged changes. It looks valid.

Approved!

**Pull request merged**